UserCode/Program.cs

```diff
@@ -1 +1,34 @@
-﻿namespace UserCode;
+﻿using System.Data.Common;
+using Examples.E3;
+CargoFactory cargoF = new CargoFactory();
+
+CargoDeliveryLocations ShipTravelLocation = new CargoDeliveryLocations
+    { 
+        Origin = "Bandar-Abbas",
+        Destination = "Qatar"
+    };
+
+
+
+cargoF.ConfigureDeliveryTypes(deliveryTypes =>
+{
+    deliveryTypes.Add(new TrainCargo());
```

Comment:
in Example I said we can not access to the source code, and we don't know how to create an instance of train or ship or air as well.

so you need to make factory method pattern. 

it is not fully compatible.

Reply:
Thanks For your attention
in Last commit:
1- Dictionary Used  instead of lists cause we dont need to iterate on that
2- ConfigureDeliveryTypes removed
3- Constructor will configure it'self
4- cargoFactory automaticly create an instance from configured classes Only when developer needed